src/develop.rs

```diff
@@ -35,9 +35,9 @@ pub fn develop(
         Ok(output) if output.status.success() => {
             let platform = String::from_utf8_lossy(&output.stdout);
             if platform.contains("macos") {
-                if platform.contains("x86_64") && target.target_arch() != Arch::X86_64 {
+                if platform.contains("x86_64") && target.target_arch() == Arch::X86_64 {
```

Comment:
L38 and L40 change is wrong. We want to respect Python architecture to fix #982 and #979. `target` is default Rust target we get from `rustc -vV`.

Reply:
> Then the offending code shouldn't be these two I think? It might be related to the `target_triple` variable, can you try #1016 ?

I've tried #1016, same error.

Reply:
But if I append `--target aarch64-apple-darwin`, it works fine.

Reply:
Try add `println!("{:#?}", build_options);` just above https://github.com/PyO3/maturin/blob/86b5dfc77254da2d01cd48a8d4128a15659d1730/src/develop.rs#L68

Reply:
```
BuildOptions {
    platform_tag: [
        Linux,
    ],
    interpreter: [
        "/Users/pvon/Documents/string_sum/.env/bin/python",
    ],
    find_interpreter: false,
    bindings: None,
    out: Some(
        "/var/folders/0r/tkx69nhn6vg6sr0w1c6nnsxr0000gn/T/.tmpAolWXa",
    ),
    skip_auditwheel: false,
    zig: false,
    universal2: false,
    cargo: CargoOptions {
        quiet: false,
        jobs: None,
        profile: None,
        features: [],
        all_features: false,
        no_default_features: false,
        target: None,
        target_dir: None,
        manifest_path: None,
        ignore_rust_version: false,
        verbose: 0,
        color: None,
        frozen: false,
        locked: false,
        offline: false,
        config: [],
        unstable_flags: [],
        timings: None,
        future_incompat_report: false,
        args: [],
    },
}
```

Reply:
Nothing seems wrong, I can't reproduce this issue, so you have to debug on your own.